lib/hammer_cli_foreman/commands.rb

```diff
@@ -519,4 +519,49 @@ def self.success_message(msg = nil)
       super(msg) || default_message(_('The %{resource_name} has been disassociated.'))
     end
   end
+
+  class DownloadCommand < HammerCLIForeman::SingleResourceCommand
+    action :download
+
+    def self.command_name(name = nil)
+      super(name) || "download"
+    end
+
+    def request_options
+      { :response => :raw }
+    end
+
+    option "--path", "PATH", _("Path to directory where downloaded content will be saved"),
+        :attribute_name => :option_path
+
+    def execute
+      response = send_request
+      if option_path
+        filepath = store_response(response)
+        print_message(_('The response has been saved to %s.') % filepath)
```

Comment:
Nitpick: I'm not going to block the PR for this, but note that you can use `print_message(_('The response has been saved to %{path}s.'), {:path => filepath})`. The benefit is that structured output adapters will print both the message and file path under separate keys -> no need to parse it if you want to use it for scripting.